src/crypto_box.rs

```diff
@@ -0,0 +1,308 @@
+//!
+//! ```
+//! // recipient has a keypair
+//! let mut pk = [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+//! let mut sk = sodoken::LockedArray::new().unwrap();
+//! sodoken::crypto_box::xsalsa_keypair(&mut pk, &mut sk.lock()).unwrap();
+//!
+//! // sender encrypts a message using receiver pub key
+//! let msg = b"test-message".to_vec();
+//! let mut cipher = vec![0; msg.len() + sodoken::crypto_box::XSALSA_SEALBYTES];
+//! sodoken::crypto_box::xsalsa_seal(&mut cipher, &msg, &pk).unwrap();
+//!
+//! // recipient can decrypt the message
+//! let mut msg = vec![0; cipher.len() - sodoken::crypto_box::XSALSA_SEALBYTES];
+//! sodoken::crypto_box::xsalsa_seal_open(&mut msg, &cipher, &pk, &sk.lock())
+//!     .unwrap();
+//!
+//! assert_eq!(
+//!     "test-message",
+//!     String::from_utf8_lossy(&msg),
+//! );
+//! ```
+//!
+//! #### Example Box Encryption / Decryption
+//!
+//! ```
+//! // recipient has a keypair
+//! let mut r_pk = [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+//! let mut r_sk = sodoken::LockedArray::new().unwrap();
+//! sodoken::crypto_box::xsalsa_keypair(&mut r_pk, &mut r_sk.lock()).unwrap();
+//!
+//! // sender has a keypair
+//! let mut s_pk = [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+//! let mut s_sk = sodoken::LockedArray::new().unwrap();
+//! sodoken::crypto_box::xsalsa_keypair(&mut s_pk, &mut s_sk.lock()).unwrap();
+//!
+//! // sender encrypts a message
+//! let msg = b"test-message".to_vec();
+//! let mut nonce = [0; sodoken::crypto_box::XSALSA_NONCEBYTES];
+//! sodoken::random::randombytes_buf(&mut nonce).unwrap();
+//! let mut cipher = vec![0; msg.len() + sodoken::crypto_box::XSALSA_MACBYTES];
+//! sodoken::crypto_box::xsalsa_easy(
+//!     &mut cipher,
+//!     &msg,
+//!     &nonce,
+//!     &r_pk,
+//!     &s_sk.lock(),
+//! ).unwrap();
+//!
+//! // receiver decrypts the message
+//! let msg_len = cipher.len() - sodoken::crypto_box::XSALSA_MACBYTES;
+//! let mut msg = vec![0; msg_len];
+//! sodoken::crypto_box::xsalsa_open_easy(
+//!     &mut msg,
+//!     &cipher,
+//!     &nonce,
+//!     &s_pk,
+//!     &r_sk.lock(),
+//! ).unwrap();
+//!
+//! assert_eq!(
+//!     "test-message",
+//!     String::from_utf8_lossy(&msg),
+//! );
+//! ```
+
+use crate::*;
+
+/// length of box seed
+pub const XSALSA_SEEDBYTES: usize =
+    libsodium_sys::crypto_box_SEEDBYTES as usize;
+
+/// length of box public key
+pub const XSALSA_PUBLICKEYBYTES: usize =
+    libsodium_sys::crypto_box_PUBLICKEYBYTES as usize;
+
+/// length of box secret key
+pub const XSALSA_SECRETKEYBYTES: usize =
+    libsodium_sys::crypto_box_SECRETKEYBYTES as usize;
+
+/// length of box mac
+pub const XSALSA_MACBYTES: usize = libsodium_sys::crypto_box_MACBYTES as usize;
+
+/// length of box nonce
+pub const XSALSA_NONCEBYTES: usize =
+    libsodium_sys::crypto_box_NONCEBYTES as usize;
+
+/// sealed box extra bytes
+pub const XSALSA_SEALBYTES: usize =
+    libsodium_sys::crypto_box_SEALBYTES as usize;
+
+#[cfg(test)]
```

Comment:
Test to bottom of file?

Reply:
Haha, woops, yeah, that was an artifact of the file I copied it from. Fixed: https://github.com/holochain/sodoken/pull/26/commits/d6694b2bdc02a2e022715347b06aa8cb17e14e6b